validator/project_validator.go

```diff
@@ -1850,178 +1804,69 @@ func validateTaskSyncCommands(p *model.Project, runLong bool) ValidationErrors {
 	return errs
 }
 
-// validateTVDependsOnTV checks that the task in the given build variant has a
-// dependency on the task in the given build variant.
-func validateTVDependsOnTV(source, target model.TVPair, tvToTaskUnit map[model.TVPair]model.BuildVariantTaskUnit) error {
-	if source == target {
-		return errors.Errorf("task '%s' in build variant '%s' cannot depend on itself",
-			source.TaskName, source.Variant)
-	}
-	visited := map[model.TVPair]bool{}
-	var allTVs []model.TVPair
-	for tv := range tvToTaskUnit {
-		visited[tv] = false
-		allTVs = append(allTVs, tv)
-	}
+// validateTVDependsOnTV checks that the dependent task always has a dependency on the depended on task.
+// The dependedOnTask and every other task along the path must run on all the same requester types as the dependentTask
+// and the dependency on the dependedOnTask must be with a status in statuses, if provided.
+func validateTVDependsOnTV(dependentTask, dependedOnTask model.TVPair, statuses []string, project *model.Project) error {
```

Comment:
I definitely prefer the original source/target naming for what it's worth -- since the variables were more different, it was easier to parse the code when reading it, whereas now you have to look closely for the "On" which separates them from each other, but that's just my 2 cents. (This also maybe separates these variables from "dependent / dependedOn" in traversal; since that's a function that iterates over everything it's hard to remember that those variable names aren't strictly related)

Reply:
I left the variable names in the function signature as dependent/depended on but renamed the variables within the function.